### PR TITLE
Fix#1203 Fix VaDataTable hooks

### PR DESCRIPTION
**Pull request by RVitaly1978**
## Description
close: #1203

Changes:
- [x] - fix useFilterable, useSelectable, useSortable
- [x] - update examples in docs

image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-data-table/hooks/useFilterable.ts [hunk 1]
```diff
@@ -25,7 +27,7 @@ export default function useFilterable (
     })
   })
 
-  watch(filteredRows, () => {
+  watch(() => filteredRows.value, () => {
```
Comment (asvae) [inline packages/ui/src/components/va-data-table/hooks/useFilterable.ts:30]: This does nothing, no?

File: packages/ui/src/components/va-data-table/hooks/useSortable.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export default function useSortable (
   // sort each time the sortBy or sortingOrder is changed (and also initially). Also if columns definitions are changed
   // (because that potentially means that the user runtime-introduced a custom sorting function for a specific column)
   watch(
-    [sortByProxy, sortingOrderProxy, columns],
+    [sortByProxy, sortingOrderProxy, () => columns.value],
```
Comment (asvae) [inline packages/ui/src/components/va-data-table/hooks/useSortable.ts:60]: Also probably does nothing?

File: packages/ui/src/components/va-data-table/hooks/useFilterable.ts [hunk 3]
```diff
@@ -1,34 +1,31 @@
...
+        ? filterMethod.value(cell.source)
+        : cell.value.toLowerCase().includes(filter.value.toLowerCase())
+    }))
   })
 
   watch(filteredRows, () => {
     const filteredRowsSource = filteredRows.value.map(row => row.source)
     emit('filtered', filteredRowsSource)
-  })
+  }, { immediate: true })
```
Comment (asvae) [inline packages/ui/src/components/va-data-table/hooks/useFilterable.ts:28]: That's probably wrong. Events should not be emitted if nothing was changed.

File: packages/ui/src/components/va-data-table/hooks/useSortable.ts [hunk 4]
```diff
@@ -81,20 +75,25 @@ export default function useSortable (
...
+  // (because that potentially means that the user runtime-introduced a custom sorting function for a specific column)
+  watch(sortedRows, () => {
     emit('sorted', {
       sortBy: sortByProxy.value,
       sortingOrder: sortingOrderProxy.value,
-      sortedRows: sortedRowsSource,
+      sortedRows: sortedRows.value.map(row => row.source),
     })
-  }
+  }, { immediate: true })
```
Comment (asvae) [inline packages/ui/src/components/va-data-table/hooks/useSortable.ts:96]: Doesn't look all right to me. Events in watch immediate.

File: packages/ui/src/components/va-data-table/hooks/useFilterable.ts (hunk 1 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-data-table/hooks/useFilterable.ts]: Fixed it.

File: packages/ui/src/components/va-data-table/hooks/useSortable.ts (hunk 2 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-data-table/hooks/useSortable.ts]: Fixed it.

File: packages/ui/src/components/va-data-table/hooks/useFilterable.ts (hunk 3 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-data-table/hooks/useFilterable.ts]: Removed and fixed.

**Pull request merged**